### DISCUSSION TITLE
Freeplay icons only show BF, do not match songs characters

### DISCUSSION THREAD

**Discussion by DragoDigitalStudios** [KadeArchive/Kade-Engine#549] [Q&A] [answered]
**Info:**
Windows x86
Version 1.4.2 of Kade Engine

Recently after adding some new songs into my mod, the icons shown on the freeplay menu no longer match that of the character in the song; the only ones that do match are the week 1 songs, but the tutorial and week 2 songs do not.
It was working until I added in the new week 2 songs.
In terms of coding:
-Songs are stated in FreeplayState.hx with the character cases
-The custom characters have properly defined icons in Healthicon.hx which work ingame
-All icons are in iconGrid.png and work ingame
-The json files for the song difficulties all match up, stating what characters are there, and all working ingame.
As far as I am aware with my coding knowledge, everything is lined up and matching fine, especially since all ingame works fine, its just these icons.

I did everything the same as I did when adding the custom songs and characters for my mod's week 1 (ADDING NEW ASSETS AND CODES, not replacing), but now this time, it has broken. 

**The issue:**
![iconsbeingwack](https://user-images.githubusercontent.com/84245123/119232041-9e57d700-bb1b-11eb-9b9b-cda04dc00ff1.png)
**All works ingame:**
![ftfrediconsandshitingame](https://user-images.githubusercontent.com/84245123/119232108-d95a0a80-bb1b-11eb-8036-4e767b4b88d7.png)
**Icons also don't line up in debug menu, even though the character of 'ftfreddy' is clearly stated (it's just using dads icon, set to be normal 'freddy')**
![debuggobrrrr](https://user-images.githubusercontent.com/84245123/119232154-07d7e580-bb1c-11eb-8e35-58c124e040ad.png)

If this is a really simple fix, then I apologise for my stupidity, but I have found nothing on how to fix it nor do I have enough coding knowledge to look too deep into the more complicated parts of the code.

**Comment by theDetourist** [comment]
Not a bug (probably), you might wanna check your freeplaySongList.txt and change the characters for the songs accordingly.

**Comment by DragoDigitalStudios** [comment]
Edit: Apparently it's not a bug, I was under the impression it was since I couldn't find any sort of reason, so my bad. Still need help on fixing it.

> Not a bug (probably), you might wanna check your freeplaySongList.txt and change the characters for the songs accordingly.

Adding the songs to freeplaySongList.txt does not fix the issue, they still have the wrong icons (also causes songs to be duplicated in the freeplay menu), plus Tutorial is already in that file and has the issue. I had already tried this.

"change the characters for the songs accordingly" if you mean in the debug menu, that's all been done for each song. If you mean in FreeplayState.hx, thats also already been done. They are also all in charactList.txt. don't know where else I could do this.

**Comment by puyoxyz** [comment, marked as answer]
Change the character in freeplaySonglist

**Reply by DragoDigitalStudios** [reply]
But how? Isn’t it just a .txt file to write song names in, not characters? How do I ‘change characters’ using this file?
(Also, I’ve changed the characters in FreeplayState.hx, if you mean that)

**Reply by puyoxyz** [reply]
> But how? Isn’t it just a .txt file to write song names in, not characters? How do I ‘change characters’ using this file?
> (Also, I’ve changed the characters in FreeplayState.hx, if you mean that)

No, it contains more information. The format is one song per line, `songname:character:week` (example: `Milf:mom:4` for a song called Milf with the character mom in week 4)

**Reply by DragoDigitalStudios** [reply]
Edit: It worked, just had to go into FreeplayState.hx and delete the songs from there to remove the duplicates. Thank you!

I can’t believe I’ve never come across that in all my research - I’ll try that tomorrow and hopefully it’ll work. Thank you.
